ccpp/driver/GFS_restart.F90

```diff
@@ -108,14 +115,22 @@ subroutine GFS_restart_populate (Restart, Model, Statein, Stateout, Sfcprop, &
     ! NoahMP
     if (Model%lsm == Model%lsm_noahmp) then
       Restart%num2d = Restart%num2d + 10
+      surface_layer_saves_rainprev = .true.
     endif
     ! RUC 
     if (Model%lsm == Model%lsm_ruc) then
       Restart%num2d = Restart%num2d + 5
+      surface_layer_saves_rainprev = .true.
```

Comment:
@SamuelTrahanNOAA Does it mean that without setting surface_layer_saves_rainprev =.true. here, there is a problem with the precipitation in the restart when MYNN surface layer is not used?

Reply:
The CLM Lake Model needs the two rain prev variables. I don't want to store them more than once though, so there's an "if" statement to store them only when the surface layer doesn't.